internal/controller/utils.go

```diff
@@ -16,6 +18,39 @@ func updateDeploymentAnnotations(deployment *appsv1.Deployment, annotations map[
 	}
 }
 
+// setDeploymentReplicas sets the number of replicas for a given deployment.
+func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int32) bool {
+	if *deployment.Spec.Replicas != replicas {
+		*deployment.Spec.Replicas = replicas
+		return true
+	}
+
+	return false
+}
+
+// setDeploymentContainerResources sets the resource requirements for a specific container in a Deployment.
+func setDeploymentContainerResources(deployment *appsv1.Deployment, resources *corev1.ResourceRequirements, containerName string) bool {
+	containerIndex := getContainerIndex(deployment, containerName)
```

Comment:
we may log the incident of missing container (containerIndex == -1) and returns.